### PR TITLE
Add madlad-400 MT models

### LINKED ISSUE #26696: MADLAD-400 MT Models
### Model description

From the paper:
> We introduce MADLAD-400, a manually audited, general domain 3T token monolingual dataset based on CommonCrawl, spanning 419 languages. We discuss the limitations revealed by self-auditing MADLAD-400, and the role data auditing had in the dataset creation process. We then train and release a 10.7B-parameter multilingual machine translation model on 250 billion tokens covering over 450 languages using publicly available data, and find that it is competitive with models that are significantly larger, and report the results on different domains. In addition, we train a 8B-parameter language model, and assess the results on few-shot translation. We make the baseline models available to the research community.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

Paper: https://arxiv.org/abs/2309.04662
Github (with checkpoint links): https://github.com/google-research/google-research/tree/master/madlad_400

### PR DESCRIPTION
**Pull request by jbochi**
# What does this PR do?

Fixes #26696 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker

### CODE REVIEW

**Comment by amyeroberts** [comment]
Thanks for adding @jbochi! Let us know when you're ready for a review.

I've set the failed tests to re-run as it seems there was a transient connection error causing them to fail.

File: docs/source/en/model_doc/madlad-400.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+Google has released the following variants:
+
+- [jbochi/madlad400-3b-mt](https://huggingface.co/jbochi/madlad400-3b-mt)
+
+- [jbochi/madlad400-7b-mt](https://huggingface.co/jbochi/madlad400-7b-mt)
+
+- [jbochi/madlad400-7b-mt-bt](https://huggingface.co/jbochi/madlad400-7b-mt-bt)
+
+- [jbochi/madlad400-10b-mt](https://huggingface.co/jbochi/madlad400-10b-mt)
```
Comment (amyeroberts) [inline docs/source/en/model_doc/madlad-400.md:45]: We'll need to transfer these under the google org on the hub and update these URLs before merging

File: docs/source/en/model_doc/madlad-400.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# MADLAD-400
+
+## Overview
+
+MADLAD-400 models were released in the paper [MADLAD-400: A Multilingual And Document-Level Large Audited Dataset](MADLAD-400: A Multilingual And Document-Level Large Audited Dataset) and converted to transformers' format by [Juarez Bochi](https://github.com/jbochi). - These are machine translation models based on the T5 architecture.
```
Comment (amyeroberts) [inline docs/source/en/model_doc/madlad-400.md:21]: To match the pattern in the rest of the docs:
* Use `This model was contributed by ...`
* Link to your huggingface hub page
* Included the paper abstract

```suggestion
MADLAD-400 models were released in the paper [MADLAD-400: A Multilingual And Document-Level Large Audited Dataset](MADLAD-400: A Multilingual And Document-Level Large Audited Dataset). 

The abstract from the paper is the following: 

*abstract here*

This model was added by  [Juarez Bochi](https://huggingface.co/jbochi). The original checkpoints can be found [here](https://github.com/google-research/google-research/tree/master/madlad_400). 

[Simple description of the model in a few sentences here]
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding! 

Just a few small notes on the doc page structure.

File: docs/source/en/model_doc/madlad-400.md (hunk 1 above)
Comment (amyeroberts) [reply docs/source/en/model_doc/madlad-400.md:45]: I can transfer them :) 

For the checkpoints - could you update their model cards to reflect other T5 models? E.g. [this one for Flan5-T5](https://huggingface.co/google/flan-t5-xxl). 

Ping once done and I'll transfer

Comment (jbochi) [reply docs/source/en/model_doc/madlad-400.md]: Thanks. I've updated the docs here and the model cards.

Can you please fix all the references to `jbochi` after you have transfer the model?

File: docs/source/en/model_doc/madlad-400.md (hunk 2 above)
Comment (jbochi) [reply docs/source/en/model_doc/madlad-400.md]: Done. PTAL.

File: docs/source/en/model_doc/madlad-400.md (hunk 1 above)
Comment (jbochi) [reply docs/source/en/model_doc/madlad-400.md]: PS: Please DO NOT transfer https://huggingface.co/jbochi/madlad400-8b-lm

The language model has a [different architecture](https://github.com/huggingface/transformers/issues/26696#issuecomment-1808181487), and I haven't figured out how to run inference yet.

**Comment by jbochi** [comment]
I think we can transfer all MT models now and merge this PR.

Maybe I will work on LM models later, but I am not sure.

What do you think?

File: docs/source/en/model_doc/madlad-400.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+The abstract from the paper is the following: 
+
+> We introduce MADLAD-400, a manually audited, general domain 3T token monolingual dataset based on CommonCrawl, spanning 419 languages. We discuss
+> the limitations revealed by self-auditing MADLAD-400, and the role data auditing
+> had in the dataset creation process. We then train and release a 10.7B-parameter
+> multilingual machine translation model on 250 billion tokens covering over 450
+> languages using publicly available data, and find that it is competitive with models
+> that are significantly larger, and report the results on different domains. In addition, we train a 8B-parameter language model, and assess the results on few-shot
+> translation. We make the baseline models 1
+> available to the research community.
```
Comment (amyeroberts) [inline docs/source/en/model_doc/madlad-400.md:32]: nit - formatting to be consistent with other model pages 

```suggestion
*We introduce MADLAD-400, a manually audited, general domain 3T token monolingual dataset based on CommonCrawl, spanning 419 languages. We discuss 
the limitations revealed by self-auditing MADLAD-400, and the role data auditing
had in the dataset creation process. We then train and release a 10.7B-parameter
multilingual machine translation model on 250 billion tokens covering over 450
languages using publicly available data, and find that it is competitive with models
that are significantly larger, and report the results on different domains. In addition, we train a 8B-parameter language model, and assess the results on few-shot
translation. We make the baseline models 1
available to the research community.*
```

**Review by amyeroberts — APPROVED** [review]
Great - thanks for iterating! 

Just one little nit on the abstract formatting. I can transfer the checkpoints, the checkpoint paths can be updated and then we can merge 🤗

**Comment by amyeroberts** [comment]
Thank you for filling out the model cards so beautifully 🙏

**Comment by jbochi** [comment]
Thanks for the review!

**Comment by amyeroberts** [comment]
Hi @jbochi. Apologies for the delay in getting back to you - I'm also learning about the intricacies of org weights too! For the model weights, would you prefer that we move the original checkpoint repos directly under google or duplicate, so that yours still exist under your profile?

**Comment by jbochi** [comment]
Hey. No worries. I think it's better to move them (assuming people will be
redirected if they go to mine).

**Comment by amyeroberts** [comment]
@jbochi If they're moved then they wouldn't exist under your profile any more at all. I believe it would error-out saying that the repo doesn't exist if someone tried: `AutoTokenizer.from_pretrained("jbochi/madlad400-3b-mt")`. Is this still OK or would you rather we copy?

**Comment by jbochi** [comment]
In this case, can we copy them instead? Thanks

**Comment by amyeroberts** [comment]
@jbochi Yep! All done. I've set the doc tests to re-run which should pass now the google checkpoints exist 🤞

**Comment by jbochi** [comment]
Yay! Thank you!

**Comment by amyeroberts** [comment]
@jbochi One last final update - the doctests are failing with a timeout. Could you add the model doc page `madlad-400.md` to the [not_doctested.txt](https://github.com/huggingface/transformers/blob/ce315081340fdf6846f16c321eb53878b6272d53/utils/not_doctested.txt#L4) page?

**Comment by amyeroberts** [comment]
@jbochi Awesome - thanks again for adding!

**Comment by jbochi** [comment]
Thanks for the review and all the help, @amyeroberts

**Pull request merged**